runbotics/runbotics-scheduler/src/mail/mail.service.ts

```diff
@@ -53,18 +57,32 @@ export class MailService {
     }
 
     public async sendBotDisconnectionNotificationMail(bot: IBot, installationId: string) {
-        await this.sendMail({
-            to: bot.user.email ?? '',
-            subject: NOTIFICATION_MAIL_SUBJECT,
-            content: `Hello,\n\nBot 🤖 (${installationId}) has been disconnected.\n\nBest regards,\nRunBotics`,
-        });
+        const disconnectedBot = await this.botService.findById(bot.id);
+
+        if (disconnectedBot.subscribers && disconnectedBot.subscribers.length) {
+            const subscribers = disconnectedBot.subscribers;
+            const emailAddresses = subscribers.map(({ email }) => email).join(',');
+
+            await this.sendMail({
+                to: emailAddresses ?? '',
+                subject: NOTIFICATION_MAIL_SUBJECT,
+                content: `Hello,\n\nBot 🤖 (${installationId}) has been disconnected.\n\nBest regards,\nRunBotics`,
+            });
+        }
     }
 
     public async sendProcessFailureNotificationMail(process: IProcess, processInstance: IProcessInstance) {
-        await this.sendMail({
-            to: process.createdBy.email ?? '',
-            subject: NOTIFICATION_MAIL_SUBJECT,
-            content: `Hello,\n\nProcess ⚙️ ${process.name} (${process.id}) has failed with status (${processInstance.status}).\n\nBest regards,\nRunBotics`,
-        });
+        const failedProcess = await this.processService.findById(process.id);
+
+        if (failedProcess.subscribers && failedProcess.subscribers.length) {
+            const subscribers = failedProcess.subscribers;
+            const emailAddresses = subscribers.map(({ email }) => email).join(',');
+
+            await this.sendMail({
+                to: emailAddresses ?? '',
+                subject: NOTIFICATION_MAIL_SUBJECT,
```

Comment:
same here

Reply:
also if emailAddresses is an empty array this will not trigger ?? so

to: emailAddresses,

is just enough